### PR TITLE
Straw-man for our workshop

### PR DESCRIPTION
**Pull request by MahrRah**
Please feel free to change, rewrite and add to these notes by committing directly into this branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,13 +4,21 @@ This is a hands-on workshop to get familiar with the observability and the relev
 
 ## Path 1: Observability of a K8s cluster
 
-This path will introduce how to instrument a K8s cluster to send telemetry to Azure Monitor dashboards.
+This path will introduce :
+
+- how to instrument applications using Opentelemtery on Azure
+- How to instrument your AKS cluster
+- How to create use your collected telemetry data by creating Azure Monitor Workbooks, Dashboards
```
Comment (Izzzu) [inline README.md:11]:
```suggestion
- How to use your collected telemetry data by creating Azure Monitor Workbooks, Dashboards
```

File: 03-create-observability-infrastructure/README.md [hunk 2]
```diff
@@ -1,2 +1,3 @@
-# Create observability infrastructure
-TBD
+# Creating Observability Infrastructure
+
+> Goal: Exploring different options to instrument standard applications with standard tooling and easily integrate them into Azure for data collection and observability.
```
Comment (Izzzu) [inline 03-create-observability-infrastructure/README.md:3]: I found some notes from our discussions with Maggie a few weeks ago and trying to applying them here

```suggestion
> Goal: Exploring different options to instrument standard applications with standard tooling and easily integrate them into Azure for data collection and observability.

In this section we are going to build the observability solution step by step by using automatic OpenTelemetry instrumentation.

1. Auto instrumentation of .Net application 
2. Auto instrumentation of Java application 
3. Extra: custom traces and metrics
 
```

File: 04-dashboards/README.md [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # Dashboards
```
[the remark this answers is not in the document]
Comment (MahrRah) [reply 04-dashboards/README.md:1]: Yes, that sounds good! Do we want to do a quick 101 on that and have deep dive on this attached in the appendix as additional material or do we want to go super deep in these folders?

File: 02-deploy-application/README.md [hunk 4]
```diff
@@ -1,3 +1,3 @@
-# Deploy Application
+# Deploying the Application
 
-TBD
+> Goal: Observe the application with basic "instrumentation," including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.
```
Comment (MaggieSalak) [inline 02-deploy-application/README.md:3]:
```suggestion
> Goal: Observe the application with basic "instrumentation", including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.
```

File: 03-create-observability-infrastructure/README.md [hunk 5]
```diff
@@ -1,2 +1,9 @@
...
```
Comment (MaggieSalak) [inline 03-create-observability-infrastructure/README.md:3]:
```suggestion
> Goal: Exploring different options to instrument applications with standard tooling and easily integrate them into Azure for data collection and observability.
```

**Review by MagdaPaj — COMMENTED** [review]
It looks very well! I would like to just discuss sections 02 and 03. But overall, I like to structure!

File: 04-dashboards/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Izzzu) [reply 04-dashboards/README.md:1]: Maybe let's align on what each option would mean. Do we want to handle the topics in the folders as a separate sections during the workshop? the 101 part would be also covered in the workshop? The appendix is more for the workshop participants to learn on their own?

File: 02-deploy-application/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Izzzu) [reply 02-deploy-application/README.md:3]: My understanding is that its any customer we have in the engagements. In the meaning that this section of the workshop allows the participants to learn how to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

[the remark this answers is not in the document]
Comment (Izzzu) [reply 02-deploy-application/README.md:3]: We could have this section as the first step for the instrumentation, meaning "deploy and see what is available out of the box in Azure", so in the first section (01) we deploy the infra: AKS, cosmosdb etc, in the second we deploy the applications into AKS and see how the system behaves and what we can see in Azure. Maybe spot what's missing which would be an introduction tot the next section that goes through the automatic instrumentations for the modules

[the remark this answers is not in the document]
Comment (MahrRah) [reply 02-deploy-application/README.md]: Hope the updated version does clarify things.  If not let me know :)

File: 02-deploy-application/README.md [hunk 6]
```diff
@@ -1,3 +1,11 @@
-# Deploy Application
+# Deploying the Application into AKS
 
-TBD
+> Goal:
+>
+> 1. Having application run full E2E and understanding the pieces.
+>
+> 2. Observe the application with basic "instrumentation" including emitted log lines and possibly some metrics using `kubectl`. Get familiar with out-of-the-box AKS cluster observability (eg. workloads, pods etc.) Mentioning of ContainerInsights and its capabilities.
```
Comment (MagdaPaj) [inline 02-deploy-application/README.md:7]:
```suggestion
> 2. Getting familiar with out-of-the-box AKS cluster observability (e.g. workloads, pods etc.) Mentioning of Container Insights and its capabilities.
```

Comment (MagdaPaj) [reply 02-deploy-application/README.md:7]: We could still show the logs through kubectl, though I would skip the first sentence and just mention out-of-the box AKS observability. Later we can review and decide what exactly we will show.

File: 02-deploy-application/README.md [hunk 7]
```diff
@@ -1,3 +1,11 @@
...
+
+1. Deploy Services to AKS
```
Comment (MagdaPaj) [inline 02-deploy-application/README.md:9]:
```suggestion
1. Deploy the application to AKS
```

File: 02-deploy-application/README.md [hunk 8]
```diff
@@ -1,3 +1,11 @@
...
+2. Create Devices and see temperature and state being updated
```
Comment (MagdaPaj) [inline 02-deploy-application/README.md:10]:
```suggestion
2. Create devices and see temperature and state being updated
```

File: 03-add-basic-observability-instrumentation/README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# Adding Basic Observability Instrumentation
+
+> Goal: Exploring different options to instrument applications with standard tooling and easily integrate them into Azure for data collection and observability.
+
+In this section we are going to build the observability solution step by step by using automatic OpenTelemetry instrumentation.
+
+1. OtelCollector introduction and ref to lower section with comparison to alertnatives
+2. Auto instrumentation of .Net application
```
Comment (MagdaPaj) [inline 03-add-basic-observability-instrumentation/README.md:8]:
```suggestion
2. Auto instrumentation of .NET application
```

File: 06-advanced-observability-instrumentation/README.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+# Advanced Observability Instrumentation
+
+> Goal: Gain insights on how to add custom metrics and traces or other specific configureations.
```
Comment (MagdaPaj) [inline 06-advanced-observability-instrumentation/README.md:3]:
```suggestion
> Goal: Gain insights on how to add custom metrics and traces or other specific configurations.
```

File: 06-advanced-observability-instrumentation/README.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
...
+
+1. How to add custom instrumentation
+2. How to add custom traces for distributed tracing
```
Comment (MagdaPaj) [inline 06-advanced-observability-instrumentation/README.md:6]:
```suggestion
2. How to add custom traces for distributed tracing
3. How to add custom labels
```

File: 10-azure-observabity-101/README.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+- Exploring data sources for each visualization tool.
+- Highlighting the pros and cons of each tool.
+
+### Dashboards
+
+### Workbooks
+
+### Third-Party Tools
+
+e.g., Grafana
```
Comment (MagdaPaj) [inline 10-azure-observabity-101/README.md:38]:
```suggestion
e.g., Grafana, Prometheus
```

File: README.md [hunk 13]
```diff
@@ -4,13 +4,23 @@ This is a hands-on workshop to get familiar with the observability and the relev
 
 ## Path 1: Observability of a K8s cluster
 
-This path will introduce how to instrument a K8s cluster to send telemetry to Azure Monitor dashboards.
+This path will introduce :
+
+- how to instrument applications using Opentelemtery on Azure
```
Comment (MagdaPaj) [inline README.md:9]:
```suggestion
- how to instrument applications using OpenTelemetry with Azure
```

File: README.md [hunk 14]
```diff
@@ -4,13 +4,23 @@ This is a hands-on workshop to get familiar with the observability and the relev
...
-- [⚒️ Pre Requisites](./00-pre-requisite/README.md) -  Covering the pre set up and tools that will be
+- [⚒️ Pre Requisites](./00-pre-requisite/README.md) - Covering the pre set up and tools that will be
   needed.
-- [⚙️ Setup Infrastructure](./01-setup-infrastructure/README.md) - Provision AKS cluster, Application Insights etc.
-- [🧩 Deploy application](./02-deploy-application/README.md) - TBD
-- [🔎 Create observability infrastructure](./03-create-observability-infrastructure/README.md) - TBD
-- [📈 Dashboards](./04-dashboards/README.md) - Visualize data with dashboards
+- [⚙️ Provision Infrastructure](./01-provision-infrastructure/README.md) - Provision AKS cluster, Application Insights etc.
+- [🧩 Deploy application to AKS](./02-deploy-application/README.md) - Deploy tbd application
+- [🔎 Add basic observability instrumentation](./03-add-basic-observability-instrumentation/README.md) - Using Opentelemetry instrument your application
```
Comment (MagdaPaj) [inline README.md:19]:
```suggestion
- [🔎 Add basic observability instrumentation](./03-add-basic-observability-instrumentation/README.md) - Using OpenTelemetry instrument your application
```

File: README.md [hunk 15]
```diff
@@ -4,13 +4,23 @@ This is a hands-on workshop to get familiar with the observability and the relev
...
 
-- [⚒️ Pre Requisites](./00-pre-requisite/README.md) -  Covering the pre set up and tools that will be
+- [⚒️ Pre Requisites](./00-pre-requisite/README.md) - Covering the pre set up and tools that will be
   needed.
-- [⚙️ Setup Infrastructure](./01-setup-infrastructure/README.md) - Provision AKS cluster, Application Insights etc.
-- [🧩 Deploy application](./02-deploy-application/README.md) - TBD
-- [🔎 Create observability infrastructure](./03-create-observability-infrastructure/README.md) - TBD
-- [📈 Dashboards](./04-dashboards/README.md) - Visualize data with dashboards
+- [⚙️ Provision Infrastructure](./01-provision-infrastructure/README.md) - Provision AKS cluster, Application Insights etc.
+- [🧩 Deploy application to AKS](./02-deploy-application/README.md) - Deploy tbd application
```
Comment (MagdaPaj) [inline README.md:18]:
```suggestion
- [🧩 Deploy application to AKS](./02-deploy-application/README.md) - Deploy all required components of the application
```

File: 00-pre-requisite/README.md [hunk 16]
```diff
@@ -1,3 +1,5 @@
 # Pre Requisites
 
-TBD
+- Azure sub
```
Comment (MagdaPaj) [inline 00-pre-requisite/README.md:3]:
```suggestion
- Azure subscription
```

**Review by MagdaPaj — APPROVED** [review]
The structure looks good! Just a bunch of small suggestions.
🚀

**Pull request merged**